### PR TITLE
Rumble ml - Implement getting and running transformers

### PR DESCRIPTION
**Pull request by CanBerker**
This PR is based on #415.

2020-01-16: First version, minimal functionality supported:
- get-transformer method for extraction transformers from SparkML API 
- Passing string parameters eg. (inputCol) is supported.
- Test case demonstrates usage

### CODE REVIEW

File: src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/base/Functions.java [hunk 1]
```diff
@@ -1905,6 +1909,16 @@ public static FunctionItem getUserDefinedFunction(FunctionIdentifier identifier)
...
 
+        /**
+         * function fetches the transformer class from SparkML API
+         */
+        static final BuiltinFunction get_transformer = createBuiltinFunction(
+            "get-transformer",
+            "string",
+            "item*", // TODO: update the return type as transformer
+            GetTransformerFunctionIterator.class
+        );
```
Comment (CanBerker) [inline src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/base/Functions.java]: Return type may be improved later on as 'Transformer'

File: src/main/java/sparksoniq/exceptions/codes/ErrorCodes.java [hunk 2]
```diff
@@ -50,6 +50,8 @@
     public static final String UnimplementedErrorCode = "RBST0002";
     public static final String JobWithinAJobErrorCode = "RBST0003";
     public static final String OurBadErrorCode = "RBST0004";
+    public static final String UnknownRumbleMLClassReferenceErrorCode = "RBST0005";
+    public static final String UnknownRumbleMLParamReferenceErrorCode = "RBST0006";
```
Comment (CanBerker) [inline src/main/java/sparksoniq/exceptions/codes/ErrorCodes.java]: @ghislainfourny Do you have a preference for Error codes & naming? 

TODO: Update error code documentation once this is settled.

File: docs/Error codes.md [hunk 3]
```diff
@@ -16,7 +16,14 @@ or arrays are supplied where an atomic element is expected.
 It is a type error if there is not exactly one supplied parameter
 for an object or array selector.
 
-- [SENR0001] - Serialization error. Function items can not be serialized
```
Comment (CanBerker) [inline docs/Error codes.md]: Moved within file

File: src/main/java/sparksoniq/exceptions/codes/ErrorCodes.java [hunk 4]
```diff
@@ -43,7 +43,9 @@
     public static final String InvalidSelectorErrorCode = "JNTY0018";
 
 
-    public static final String FunctionsNonSerializable = "SENR0001";
```
Comment (CanBerker) [inline src/main/java/sparksoniq/exceptions/codes/ErrorCodes.java]: Moved within file

File: src/main/java/sparksoniq/exceptions/codes/ErrorCodes.java (hunk 2 above)
Comment (ghislainfourny) [reply src/main/java/sparksoniq/exceptions/codes/ErrorCodes.java]: Discussed offline, all clear

**Pull request merged**